module/VuFindTheme/src/VuFindTheme/ResourceContainer.php

```diff
@@ -129,12 +129,130 @@ public function addCss($css)
+    {
+        if (!is_array($jsEntry)) {
+            $this->addJsStringEntry($jsEntry);
+        } else {
+            $this->addJsArrayEntry($jsEntry);
         }
-        foreach ($js as $current) {
-            $this->js[] = $current;
+    }
+
+    /**
+     * Helper function for adding a Javascript file which is described as string.
+     *
+     * @param string $jsEntry Entry to add as string.
+     *
+     * @return void
+     */
+    protected function addJsStringEntry($jsEntry)
+    {
+        $parts = $this->parseSetting($jsEntry);
+        if (count($parts) == 1) {
+            $jsEntry = ['file' => $jsEntry];
+        } else {
+            $jsEntry = [
+                'file' => $parts[0],
+                'attributes' => ['conditional' => trim($parts[1])],
+            ];
         }
+        $this->addJsArrayEntry($jsEntry);
+    }
+
+    /**
+     * Helper function for adding a Javascript file which is described as array.
+     *
+     * @param string $jsEntry Entry to add as string.
+     *
+     * @return void
+     */
+    protected function addJsArrayEntry($jsEntry)
+    {
+        if (isset($jsEntry['priority']) && isset($jsEntry['load_after'])) {
+            throw new \Exception(
+                'Using "priority" as well as "load_after" in the same entry '
+                . 'is not supported: "' . $jsEntry['file'] . '"'
+            );
+        }
+
+        foreach ($this->js as $existingEntry) {
+            if ($existingEntry['file'] == $jsEntry['file']) {
+
+                // If we have the same settings as before, just skip this entry.
+                if (count(array_diff($existingEntry, $jsEntry)) > 0) {
+                    return;
+                }
+
+                throw new \Exception(
+                    'Overriding an existing dependency is not supported: '
+                    . '"' . $jsEntry['file'] . '"'
+                );
+            }
+        }
+
+        $this->insertEntry($jsEntry, $this->js);
+    }
+
+    /**
+     * Helper function to insert an entry to an array,
+     * also considering priority and dependency, if existing.
+     *
+     * @param array $entry The entry to insert.
+     * @param array $array The array into which the entry shall be inserted.
+     *
+     * @return void
+     */
+    protected function insertEntry($entry, &$array)
+    {
+        if (isset($entry['priority']) || isset($entry['load_after'])) {
+            foreach (array_keys($array) as $i) {
+                if (isset($entry['priority'])) {
+                    $currentPriority = $array[$i]['priority'] ?? null;
+                    if (!isset($currentPriority)
+                        || $currentPriority > $entry['priority']
+                    ) {
+                        array_splice($array, $i, 0, [$entry]);
+                        return;
+                    }
+                } elseif (isset($entry['load_after'])) {
+                    if ($entry['load_after'] == $array[$i]['file']) {
```

Comment:
Could use `===`?

Reply:
Right, strict comparison can be used here.